src/main/java/org/opensrp/web/controller/UniqueIdController.java

```diff
@@ -123,11 +137,26 @@ boolean checkRoleIfRoleExitst(List<String> roleList, String role) {
 		
 		String numberToGenerate = getStringFilter("numberToGenerate", request);
 		String source = getStringFilter("source", request);
+		Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
```

Comment:
could you pass authentication as param to this method for easier unit testing of this later